compiler/src/dotty/tools/dotc/interactive/Completion.scala

```diff
@@ -302,6 +302,7 @@ object Completion {
     def selectionCompletions(qual: Tree)(using Context): CompletionMap =
       implicitConversionMemberCompletions(qual) ++
         extensionCompletions(qual) ++
+        matchTypeCompletions(qual) ++
```

Comment:
I think this shouldn't be a separate case but we should rather ensure that match types used as method receivers are properly reduced in all other cases. We should also add test cases for them. For some reason in case of extension methods surprisingly this seemed to work even without this PR but still we should add a test to avoid regressions. We should also make sure this works well with implicit conversions.